### DISCUSSION TITLE
Score timewise (deprecate?)

### DISCUSSION THREAD

**Discussion by nrei1950** [w3c-cg/musicxml#483] [Q&A]
I'm working with files that represent music in a time wise fashion.  But I've unable to find sample file to use as guide for creating timewise music xml files.  I've tried using parttime.sxl, but all that happens is I get a message saying "DTD is phohibited".  Can someone offer me some help in getting an example of a score encoded using  timewise 
encoding?  Or help getting the xsl transformation to do the conversion?

**Comment by infojunkie** [comment]
I was able to use `xsltproc` to transform an arbitrary MusicXML file using `parttime.xsd` and back using `timepart.xsd`. It did give  me an error saying:
```
error : Unknown IO error
here.xml:2: warning: failed to load external entity "http://www.musicxml.org/dtds/timewise.dtd"
dare//DTD MusicXML 4.0 Timewise//EN" "http://www.musicxml.org/dtds/timewise.dtd"
```
but the output xml was still produced. In the past, I've had to patch the various xsd files shipped here to avoid warnings as per [this blog post](https://blog.karimratib.me/2020/11/17/validate-musicxml.html).

**Reply by Jojo-Schmitz** [reply]
See #480

**Comment by mscuthbert** [comment]
I'm thinking of marking the Timewise format as deprecated in v4.1 -- it was a great idea at the beginning to have a format that could represent both major ways of representing music (ca. 2000) but in practice, tools for working with timewise scores and timewise scores themselves are very few in the MusicXML ecosystem, and I wonder if the support costs for timewise are worth it given that partwise has essentially won as an interchange format (and work on a format for internal representation continues in MNX not MusicXML)

**Reply by nrei1950** [reply]
This makes my task even harder.  I need to  know how to create timewise scores.  So I can then use xsl to transform them to the partwise format.  Can ANYBODY please explain how to get  [ast the "DTD is prohibited"  message?

**Reply by infojunkie** [reply]
As per my answer above, I showed that using `xsltproc`, I _am_ able to perform either transformation and obtain output. Hope that helps.

**Reply by samuelbradshaw** [reply]
You could try changing
http://www.musicxml.org/dtds/timewise.dtd 
to the latest Wayback Machine cached version:
https://web.archive.org/web/20220119055857id_/https://www.musicxml.org/dtds/timewise.dtd

**Comment by nrei1950** [comment]
I finally got xsltproc to work.  It turns out that some of the xml sample files produce errors, whereas other produce only warnings.  The ones that produced warnings worked.  The results were disapponting.  I can see why no one uses it.  Instead of timewise, it should be called measurewise.  It is simply partwise by measure.  I thought it would be note by note instead of notes per measure, i.e.,  all the voices that sound at the same time would be coded before moving on to the next note.

I would like to suggest a new tag for version 4.1.  I call it voicenote -- I'm sure there is a better name for this tag.  Just as the tag </chord> indicates that the current note belongs on the same stem of the previous note, </voicenote> would indicate that this note sounds simultaneously with the previous note.  This eliminates the need for the backwards and forward tags.  Not knowing the internal data structures of any music processing programs, I don't know how easy or difficult this new tag would be to implement.

**Reply by mscuthbert** [reply]
Thanks for the comments -- at this point, I don't see a possibility of removing forward and backward in MusicXML, but do note that these have been problematic tags (especially backward) and the discussion on MNX is to remove them in future notation types.

I do wish that the xml sample files just worked straight out, but yes, as you conclude, this is one of the reasons why Timewise is considered for deprecation.

**Comment by DavidMWebber** [comment]
I'd be very happy for 'timewise' to be dropped.   I've never found a file of that format and my software (https:///www.mozart.co.uk) doesn't support it.

**Comment by mscuthbert** [comment]
With no objections over the past 5 months, let's consider the Timewise format of MusicXML deprecated in MusicXML 4.1.
